hazelcast/src/test/java/com/hazelcast/util/FilteringClassLoader.java

```diff
@@ -96,6 +95,28 @@ public InputStream getResourceAsStream(String name) {
         return delegatingClassLoader.loadClass(name);
     }
 
+    public boolean checkResourceExcluded(String resourceName) {
+        if (resourceName == null) {
+            return true;
+        }
+        // transform resource path as class name so we can check against excluded packages
+        String resourceAsClassName = resourceName.replace('/', '.').replace(".class", "");
+        for (String excludedPackage : excludePackages) {
+            if (resourceAsClassName.startsWith(excludedPackage)) {
```

Comment:
if I want to exclude package `domain.map` but not `domain.mapreduce` this will block both packages.

Reply:
Yes, the `FilteringClassLoader` has always been matching the prefix. This PR is not about changing the matching logic but about applying it consistently so if the `FilteringClassLoader` is supposed to exclude class `domain.A` from being loaded, then it should neither allow obtaining `domain/A.class` as a resource.